rules/builds.smk

```diff
@@ -141,43 +141,81 @@ rule mask:
             --output {output.alignment} 2>&1 | tee {log}
         """
 
-def _get_group_by_wildcards(wildcards):
-    region_key = f"group_by_{wildcards.region}"
-
-    if region_key in config["subsample_focus"]:
-        return config["subsample_focus"][region_key]
+def _get_subsampling_settings(wildcards):
+    # Allow users to override default subsampling with their own settings keyed
+    # by location type and name. For example, "region_europe" or
+    # "country_iceland". Otherwise, default to settings for the location type.
+    subsampling_scheme = _get_subsampling_scheme_by_build_name(wildcards.build_name)
+    subsampling_settings = config["subsampling"][subsampling_scheme]
+
+    if hasattr(wildcards, "subsample"):
+        return subsampling_settings[wildcards.subsample]
     else:
-        return config["subsample_focus"]["group_by"]
+        return subsampling_settings
+
+
+def get_priorities(wildcards):
+    subsampling_settings = _get_subsampling_settings(wildcards)
 
-def _get_sequences_per_group_by_wildcards(wildcards):
-    if wildcards.region == "global":
-        return config["subsample_focus"]["seq_per_group_global"]
+    if "priorities" in subsampling_settings and subsampling_settings["priorities"]["type"] == "proximity":
+        return f"results/{wildcards.build_name}/proximity_{subsampling_settings['priorities']['focus']}.tsv"
     else:
-        return config["subsample_focus"]["seq_per_group_regional"]
+        # TODO: find a way to make the list of input files depend on config
+        return config["files"]["include"]
 
-def _get_focus_exclude_argument_by_wildcards(wildcards):
-    if wildcards.region == "global":
-        return ""
+
+def get_priority_argument(wildcards):
+    subsampling_settings = _get_subsampling_settings(wildcards)
+
+    if "priorities" in subsampling_settings and subsampling_settings["priorities"]["type"] == "proximity":
+        return "--priority " + get_priorities(wildcards)
     else:
-        return f"--exclude-where \"region!={wildcards.region.replace('-', ' ')}\""
+        return ""
+
+
+def _get_specific_subsampling_setting(setting, optional=False):
+    def _get_setting(wildcards):
+        subsampling_scheme = _get_subsampling_scheme_by_build_name(wildcards.build_name)
```

Comment:
`subsampling_scheme` isn't used here (as far as I can tell)

Reply:
you are right. I deleted this.